Updating content that storage is always partitioned (#24250)
mdn/content: files/en-us/web/api/storage_access_api/index.md

{{DefaultAPISidebar("Storage Access API")}}

> The **Storage Access API** provides a way for embedded, cross-origin content to gain unrestricted access to the cookies that it would typically only have access to in a first-party context (We refer to storage and cookies that an origin has access to when being in a first-party context as the origin's _first-party_ storage).

Reviewer: ```suggestion
The **Storage Access API** provides a way for embedded, cross-origin content to gain unrestricted access to the cookies that it would typically only have access to in a first-party context. (We refer to storage and cookies that an origin has access to when being in a first-party context as the origin's _first-party_ storage).
```

> > **Note:** The [browser compatibility section](#browser_compatibility) provides an overview about the support staus of the Storage Access API.

Reviewer: This not is not needed because "Browser compatibility" is a standard section on the reference pages

```suggestion

```

Author: My changes to the storage_access_api doc are obsolete now, there is already an update in main that took care of the issues I am addressing in this pull request.

{{DefaultAPISidebar("Storage Access API")}}

> The Storage Access API provides a way for embedded, cross-origin content to gain unrestricted access to the cookies that it would normally only have access to in a first-party context (We refer to storage and cookies that an origin has access to when being in a first party contex as the origin's _first-party_ storage).

Author: My changes to this doc are obsolete since there is a change in main that takes care of what I have bee addressing with my changes.

In its latest version the Storage Access API can only be used to gain access to first-party cookies. Firefox adopts this behavior starting from version 109 and keeps (non-cookie) storage permanently partitioned.

The API provides methods that allow embedded resources to check whether they currently have access to their first-party cookies, and to request access to them from the user agent.

## Concepts and usage

Most browsers implement a number of storage access policies that restrict access to cookies and site data for embedded, cross-origin resources. These restrictions range from giving embedded resources under each top-level origin a unique storage space to outright blocking of storage access when resources are loaded in a third-party context.

The semantics around third-party cookie blocking policies in particular differ from browser to browser, but the core functionality is similar: cross-origin resources embedded in a third-party context are not given access to the same cookies and site storage that they would have access to when loaded in a first-party context.

> These cookie blocking policies are known to break embedded cross-origin content that requires access to its first-party storage. As an example, federated logins often require access to authentication cookies available in a first-party context, and will require the user to sign in on each site separately (or completely break) if those cookies are not available. In the case of breakage, site owners have often encouraged users to add their site as an exception or to disable the policy entirely. As a consequence, users who wish to continue to interact with embedded content are forced to greatly relax their blocking policy for resources loaded from all embedded origins and possibly across all websites.

Reviewer 2: ```suggestion
These cookie-blocking policies are known to break embedded cross-origin content that requires access to its first-party storage. As an example, federated logins often require access to authentication cookies available in a first-party context, and will require the user to sign in on each site separately (or altogether break) if those cookies are not available. In the case of breakage, site owners have often encouraged users to add their site as an exception or to disable the policy entirely. Consequently, users who wish to continue interacting with embedded content must relax significantly their blocking policy for resources loaded from all embedded origins and possibly across all websites.
```
